src/Platforms/AbstractPlatform.php

```diff
@@ -1550,15 +1551,30 @@ public function getCreateTableSQL(Table $table, $createFlags = self::CREATE_INDE
         $options['indexes']           = [];
         $options['primary']           = [];
 
-        if (($createFlags&self::CREATE_INDEXES) > 0) {
+        if (($createFlags & self::CREATE_INDEXES) > 0) {
             foreach ($table->getIndexes() as $index) {
                 /** @var $index Index */
-                if ($index->isPrimary()) {
-                    $options['primary']       = $index->getQuotedColumns($this);
-                    $options['primary_index'] = $index;
-                } else {
+                if (! $index->isPrimary()) {
                     $options['indexes'][$index->getQuotedName($this)] = $index;
+
+                    continue;
                 }
+
+                $options['primary']       = $index->getQuotedColumns($this);
+                $options['primary_index'] = $index;
+            }
+
+            foreach ($table->getUniqueConstraints() as $uniqueConstraint) {
+                /** @var UniqueConstraint $uniqueConstraint */
```

Comment:
Given the type of the collection being iterated over, is this type hint needed?

https://github.com/doctrine/dbal/blob/de1f02b628082bf3956d1d64941b4480efc81be1/src/Schema/Table.php#L710-L712